Adding a host alias fetch from the kubelet

jiquiame opened the pull request:
### What does this PR do?

Making the host alias logic fetching the hostname used by kubernetes.

### Motivation

So that the metrics from kubernetes_state no longer create duplicated hosts as the host alias from kubernetes is different from the ones that are already fetched.

Commit:
Adding a host alias fetch from the kubelet

Commit:
Change debug log

antoinepouille commented on pkg/util/kubernetes/hostinfo/host_alias.go:
```diff
@@ -0,0 +1,26 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubelet,kubeapiserver
```
👍 Thanks for the details about build tags!

antoinepouille commented on pkg/util/kubernetes/hostinfo/host_alias.go:
```diff
@@ -0,0 +1,26 @@
+
+// GetHostAlias uses the "kubelet" hostname provider to fetch the kubernetes alias
+func GetHostAlias() (string, error) {
+	if getKubeletHostname, found := hostname.ProviderCatalog["kubelet"]; found {
```
Yep, no need to use the ProviderCatalog here. I was worried with other cyclic dependancies, but we're good here

Commit:
Change dependancies + build tags

Commit:
Releasenote squash for ci

adding dummy function when k8s is not built, for puppy agent

added release note

fix reno note location

Added details to the reno note

fix reno note + formatting

xvello reviewed (approved):
LGTM, let's get croissant to confirm that should not bring any side-effects

olivielpeau reviewed (approved):
👍 LGTM as long as there's a timeout (cumulatively < 1sec) on any network call that `kubelet.HostnameProvider` might make :)

In general adding host aliases is pretty safe as long as they're actually unique to each host
